src/kubectl-application-shell

```diff
@@ -80,69 +76,32 @@ then
     fi
 fi
 
-if [[ "$image" == "" ]]
-then
-    echo "Missing image. Using the first currently deployed image from the deployment name."
-    image=$(kubectl get deployment --namespace="$namespace" "$name" -o=jsonpath='{$.spec.template.spec.containers[:1].image}')
-fi
-
-if [[ ! "$image" =~ ":" ]]
-then
-    if [[ $image == "" ]]
-    then
-        echo "Missing explicit image tag."
-        exit 1
-    fi
-fi
-
-if [[ "$config" == "" ]]
-then
-    echo "Missing config map. Using the first currently deployed config-map from the deployment name."
-    config=$(kubectl get deployment --namespace="$namespace" "$name" -o=jsonpath='{$.spec.template.spec.containers[:1].envFrom[0].configMapRef.name}')
-fi
-
-if [[ "$secret" == "" ]]
-then
-    echo "Missing secret map. Using currently deployed secret-map from the deployment name, if there is one."
-    secret=$(kubectl get deployment --namespace="$namespace" "$name" -o=jsonpath='{$.spec.template.spec.containers[:1].envFrom[0].secretRef.name}')
-fi
-
-overrides_start="{
+deploymentInfo=$(kubectl get deployment --namespace="$namespace" "$deployment" -o json)
+image=$(echo "${deploymentInfo}" | jq '.spec.template.spec.containers[0].image')
+containerName=$(echo "${deploymentInfo}" | jq '.spec.template.spec.containers[0].name')
+env=$(echo "${deploymentInfo}" | jq '.spec.template.spec.containers[0].env')
+envFrom=$(echo "${deploymentInfo}" | jq '.spec.template.spec.containers[0].envFrom')
+annotations=$(echo "${deploymentInfo}" | jq '.spec.template.metadata.annotations')
+
+overrides=$(echo "{
+  \"metadata\": {
+    \"annotations\": ${annotations}
+  },
```

Comment:
- There is an important reason we need an override for image, config and secret maps. For instance, you may wish to use a monolithic image, and use another secret and config map. 
- Annotations however, we should definitely add. 
- Storing the deployment information once and reusing it is a good idea.

Reply:
Could you please elaborate a little bit more on the reason behind these 3 parameters?

From what I understand of the purpose of the tool is to reproduce a shell from an already running application, replicating the same environment but not interfering with the existing environment.

IMO, supporting these parameters is kinda against the definition of the tool (Kubectl plugin that spins up a container with image, config & secret mapping matching an existing deployment.).

I think for these other usage cases, you already have `kubectl run`.

Reply:
I am referring to overrides to mimic Kubernetes jobs, which don't have a deployment.